analyzers/src/SonarAnalyzer.CSharp/Rules/MemberInitializerRedundant.cs

```diff
@@ -63,168 +61,178 @@ public sealed class MemberInitializerRedundant : SonarDiagnosticAnalyzer
                         return;
                     }
 
-                    var constructorSymbols = members.OfType<IMethodSymbol>().Where(IsExplicitlyDefinedConstructor).ToList();
-                    ReportRedundantInitializers(c,
-                        declaration,
-                        constructorSymbols,
-                        x => !x.Modifiers.Any(IsStaticOrConst),
-                        x => !x.Modifiers.Any(IsStaticOrConst));
-
-                    var staticInitializerSymbols = members.OfType<IMethodSymbol>().Where(IsStaticInitializer).ToList();
-                    if (staticInitializerSymbols?.Count > 0)
-                    {
-                        var staticInitializerDeclarations = GetInitializerTuples<BaseMethodDeclarationSyntax>(c, staticInitializerSymbols);
```

Comment:
[refactoring help] most of these calls have been moved in the `InitializationVerifierFactory`